main.go

```diff
@@ -90,55 +97,74 @@ func main() {
 	fmt.Println()
 	log.Infof("Deploying files")
 
-	publicInstallPages, err := deploy(clearedFilesToDeploy, config)
+	artifactURLCollection, err := deploy(clearedFilesToDeploy, config)
 	if err != nil {
 		fail("%s", err)
 	}
 	fmt.Println()
 	log.Donef("Success")
 	log.Printf("You can find the Artifact on Bitrise, on the Build's page: %s", config.BuildURL)
 
-	if err := exportInstallPages(publicInstallPages, config); err != nil {
+	if err := exportInstallPages(artifactURLCollection, config); err != nil {
 		fail("%s", err)
 	}
 	deployTestResults(config)
 }
 
-func exportInstallPages(publicInstallPages map[string]string, config Config) error {
-	if len(publicInstallPages) > 0 {
-		temp := template.New("Public Install Page template")
-		var pages []PublicInstallPage
-		for file, url := range publicInstallPages {
-			pages = append(pages, PublicInstallPage{
-				File: file,
-				URL:  url,
-			})
-		}
+func exportInstallPages(artifactURLCollection ArtifactURLCollection, config Config) error {
+	if len(artifactURLCollection.PublicInstallPageURLs) > 0 {
+		pages := mapURLsToInstallPages(artifactURLCollection.PublicInstallPageURLs)
 
 		if err := tools.ExportEnvironmentWithEnvman("BITRISE_PUBLIC_INSTALL_PAGE_URL", pages[0].URL); err != nil {
 			return fmt.Errorf("failed to export BITRISE_PUBLIC_INSTALL_PAGE_URL, error: %s", err)
```

Comment:
`failed to do magic, error: no magic` is not a recommended pattern. Rather use `failed to do magic: no magic`. 
Like in https://godoc.org/github.com/pkg/errors#Wrap
```suggestion
			return fmt.Errorf("failed to export BITRISE_PUBLIC_INSTALL_PAGE_URL: %s", err)
```

Please apply below as well.

Reply:
Not fair! Hasn't even touched that :P